clerk.go

```diff
@@ -0,0 +1,363 @@
+// Package clerk provides a way to communicate with the Clerk API.
+// Includes types for Clerk API requests, responses and all
+// available resources.
+package clerk
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+	"sync"
+	"time"
+)
+
+const (
+	sdkVersion      string = "v2.0.0"
+	clerkAPIVersion string = "v1"
```

Comment:
I believe this is not needed anymore, given our plan for API versioning.

Reply:
:+1: When we're done with API versioning I'll revisit. Leaving this mostly as a placeholder for now.